tasks/core/apps.yml

```diff
@@ -38,143 +38,104 @@
         ]
       }}
 
-- name: Install external apps
-  block:
-    - name: Find full nextcloud version
-      command: ./occ status  --output=json
-      args:
-        chdir: "{{ nextcloud_installation_dir }}"
-      become: true
-      become_user: "{{ nextcloud_file_owner }}"
-      register: _result
-      changed_when: false
-
-    - name: Remove non-json text from command output
-      set_fact:
-        nextcloud_full_version:
-          "{{ (_result.stdout | from_json)['versionstring'] }}"
-
-    - name: Create app json file url
-      set_fact:
-        nextcloud_app_store_json_url: >-
-          https://apps.nextcloud.com/api/v1/platform/{{
-            nextcloud_full_version
-          }}/apps.json
-      changed_when: false
-
-    - name: Download the app json file
-      get_url:
-        url: "{{ nextcloud_app_store_json_url }}"
-        dest: /tmp
-      register: result
-      until: result is success
+# Remove all apps from the list of external apps which
+# - is not in the configured list of apps
+# - is in the configured list of apps but have their state set to "absent"
+- name: Remove external apps
+  command: php occ app:remove "{{ item }}"
+  with_items: "{{ (nextcloud_installed_apps.enabled |
+    combine(nextcloud_installed_apps.disabled)) }}"
+  become: true
+  become_user: "{{ nextcloud_file_owner }}"
+  args:
+    chdir: "{{ nextcloud_installation_dir }}"
+  register: result
+  failed_when: result.stdout is not search('removed') or result is failed
+  changed_when: result is not failed
+  when: >-
+    ((nextcloud_apps | selectattr('name', 'search', item) | list | first
+    | default({'state':'absent'})).state | default('enabled')) == 'absent'
```

Comment:
My concern is that the role will mess up with manual actions of the user. I think we should only do what the `nextcloud_apps` dict says. What if we looped only the `nextcloud_apps` array?

Reply:
I see your point and was also wondering about how to do this best.

The issue with your suggestion is that we then have no way to un-install apps which have been removed from the `nextcloud_apps` list but they will stay there forever.

Maybe we could add a new parameter to select whether to keep manual changes in the instance or not?